lib/NodeStuffPlugin.js

```diff
@@ -56,6 +56,14 @@ class NodeStuffPlugin {
 								dep.loc = expr.loc;
 								parser.state.module.addPresentationalDependency(dep);
 							});
+					} else {
+						parser.hooks.expression
+							.for("global")
+							.tap("NodeStuffPlugin", expr => {
+								const dep = new ConstDependency("undefined", expr.range);
+								dep.loc = expr.loc;
+								parser.state.module.addPresentationalDependency(dep);
+							});
```

Comment:
I think we should throw a warning, also same for `__dirname` and `__filename`, also change it on undefined is breaking change, some developers can rely on this behavior

Reply:
yep,  you are correct